server/store/management/commands/create_new_product.py

```diff
@@ -16,37 +16,55 @@
 
 
 import pathlib
-from django.core.management.base import BaseCommand, CommandParser
 
+from django.core.management import call_command
+from django.core.management.base import BaseCommand, CommandParser
+from django.db import IntegrityError
 from store.models import Product
 
 
 class Command(BaseCommand):
     help = "Creates new active product, deactivating existing active product"
 
     def add_arguments(self, parser: CommandParser) -> None:
-
         parser.add_argument("--name", default="Name", type=str)
         parser.add_argument("--description", default="Description", type=str)
         parser.add_argument("--price", default="0", type=float)
         parser.add_argument("--discount_percent", default="0", type=int)
         parser.add_argument("--inventory_count", default="0", type=int)
         parser.add_argument("--image", type=str)
+        parser.add_argument("--testimonials", default="0", type=int)
 
     def handle(self, *args, **options):
-        # Deacivate any existing active products
-        Product.objects.filter(active=True).update(active=False)
+        name = options["name"]
 
-        product = Product.objects.create(
-            name=options["name"],
+        # nicely handle uniqueness constraints
+        product, created = Product.objects.get_or_create(
+            name=name,
             description=options["description"],
             price=options["price"],
             active=True,
             discount_percent=options["discount_percent"],
             inventory_count=options["inventory_count"],
         )
 
-        if options["image"]:
-            image = open(options["image"], "rb")
-            file_ext = pathlib.Path(options["image"]).suffix
-            product.image.save(str(product.id) + "_image" + file_ext, image)
+        if created:
+            # Deactivate any existing active products, except for this one.
```

Comment:
question: I assume "deactivate" here  means excluding other active products for purposes of testimonial creation, so that we can create testimonials for one product at a time? The word deactivate sounded like maybe we're modifying product status in the database but that's not what I'm seeing in the code.

Reply:
The "active" product is the one that's displayed on the homepage. The CUJ here was roughly "When I add a new product to my store, it takes primary placement". This has no affect on testimonial creation.